template/web/wp-content/themes/studiometa/app/Repositories/PostTypeRepository.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Repository entity for retrieving post type objects.
+ *
+ * @package Studiometa
+ */
+
+namespace Studiometa\Repositories;
+
+/** Class */
+class PostTypeRepository extends Repository {
+	const POST_TYPES = array( 'post' ); // Main post types.
+
+	/**
+	 * Returns a chronological list of latest "Post" (articles) posts for a given
+	 * category. Default $limit is 10.
+	 *
+	 * @param string|array $slug    Slug.
+	 * @param integer      $limit   Number to return (optional).
+	 * @param array        $exclude Posts to exclude (optional).
+	 * @param integer      $paged   Enable pagination (optional).
+	 *
+	 * @return Repository
+	 */
+	public function posts_by_category_slug( $slug, $limit = 10, $exclude = array(), $paged = 0 ) {
+
+		// Set sane defaults so we don't do full table scans.
+		if ( $limit <= 0 || $limit > 100 ) {
```

Comment:
Maybe throw an error or warning if the `$limit` parameter is bigger than 100 to let the developer know it should not be done?

```php
if ( $limit <= 0 || $limit > 100 ) {
  trigger_error( 'Some message to warn...', E_USER_WARNING );
  $limit = 100;
}